logger/logger.go

```diff
@@ -40,35 +40,52 @@ const (
 
 var level logging.Level
 var isWindows bool
+var customLogger CustomLogger
+
+type CustomLogger interface {
+	Log(logLevel logging.Level, msg string)
+}
+
+func SetCustomLogger(l CustomLogger) {
+	customLogger = l
+}
 
 // Infof logs INFO messages
 func Infof(msg string, args ...interface{}) {
 	GaugeLog.Infof(msg, args...)
```

Comment:
perhaps `write` should also include writing to `GaugeLog`.

Reply:
We write to GaugeLog with different log levels. GaugeLog.Infof, GaugeLog.Errorf, etc
Writing to GaugeLog in write method would introduce another switch case.